### PR TITLE
feat: [SRM-12324]: Removing validation for single time-series in AWS CloudWatch sample data fetch API response. Updating corresponding UT.

### PR DESCRIPTION
**Pull request by dhruv-harness**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38469)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38469)

File: 300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/CloudWatchServiceImpl.java [hunk 1]
```diff
@@ -68,11 +68,7 @@ public Map fetchSampleData(ProjectParams projectParams, String connectorIdentifi
 
       final Gson gson = new Gson();
```
Comment (n00bitax) [inline 300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/CloudWatchServiceImpl.java:69]: do we need to instantiate Gson locally here , why are we not using a static instance like objectmapper from JsonUtils ?

Comment (dhruv-harness) [reply 300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/CloudWatchServiceImpl.java]: Addressed

**Comment by AbhijithHarness** [comment]
*[`300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/CloudWatchServiceImpl.java` line 65 at r3](https://reviewable.io/reviews/harness/harness-core/38469#-NDlGAHK-KdEWhI_XEBS:-NDlGAHK-KdEWhI_XEBT:bwezwyy) ([raw file](https://github.com/harness/harness-core/blob/92d5767843603e91e252bad271d14f4375adefa3/300-cv-nextgen/src/main/java/io/harness/cvng/core/services/impl/CloudWatchServiceImpl.java#L65)):*
> ```Java
>       OnboardingResponseDTO response =
>           onboardingService.getOnboardingResponse(projectParams.getAccountIdentifier(), onboardingRequestDTO);
>       return JsonUtils.asMap(JsonUtils.asJson(response.getResult()));
> ```

nit: We maynot need JsonUtils at all, we should be able to return the response.getResult as it is. You can check it and fix in next PR.



_Code quote:_
```Java
JsonUtils.asMap
```

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38469)* status: 1. unresolved discussions.

**Review by n00bitax — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38469)* status: 1. unresolved discussions.

**Review by n00bitax — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38469)* status: :shipit: complete! ✓

**Pull request merged**